File: Backend/renderer/src/main/java/net/chrisparton/sparkled/renderer/fill/FillFunction.java
package net.chrisparton.sparkled.renderer.fill;

import net.chrisparton.sparkled.model.animation.effect.Effect;
import net.chrisparton.sparkled.model.animation.fill.Fill;
import net.chrisparton.sparkled.model.render.Led;
import net.chrisparton.sparkled.renderer.context.RenderContext;

public interface FillFunction {

    /**
     * Fill the provided {@link Led} using the {@link Fill} configuration of the provided {@link Effect}.
     */
    void fill(RenderContext ctx, Led led, float alpha);
}

File: renderer/src/main/kotlin/io/sparkled/renderer/fill/FillFunction.kt
package io.sparkled.renderer.fill

import io.sparkled.model.animation.effect.Effect
import io.sparkled.model.animation.fill.Fill
import io.sparkled.renderer.context.RenderContext
import java.awt.Color

interface FillFunction {

    /**
     * @return the fill color for a given LED using the [Fill] configuration of the provided [Effect].
     */
    fun getFill(ctx: RenderContext, ledIndex: Int): Color
}
